test_runner/batch_others/test_wal_acceptor_async.py

```diff
@@ -404,3 +404,55 @@ def test_concurrent_computes(neon_env_builder: NeonEnvBuilder):
 
     env.neon_cli.create_branch('test_concurrent_computes')
     asyncio.run(run_concurrent_computes(env))
+
+
+# Stop safekeeper and check that query cannot be executed while safekeeper is down.
+# Query will insert a single row into a table.
+async def check_unavailability(sk: Safekeeper,
+                               conn: asyncpg.Connection,
+                               key: int,
+                               start_delay_sec: int = 2):
+    # shutdown one of two acceptors, that is, majority
+    sk.stop()
+
+    bg_query = asyncio.create_task(conn.execute(f"INSERT INTO t values ({key}, 'payload')"))
+
+    await asyncio.sleep(start_delay_sec)
+    # ensure that the query has not been executed yet
+    assert not bg_query.done()
```

Comment:
I'm not very familiar with python async: is the task ever "done", if you don't call wait on it?

Reply:
I'm also not an expert, but `create_task` should start it in the background.

Reply:
You could comment out the `sk.stop()` above, and verify that the assertion fails. Or just use `wait` to wait for both tasks concurrently, to be safe.

Reply:
https://docs.python.org/3/library/asyncio-task.html#asyncio.create_task

> Wrap the coro [coroutine](https://docs.python.org/3/library/asyncio-task.html#coroutine) into a [Task](https://docs.python.org/3/library/asyncio-task.html#asyncio.Task) and schedule its execution. Return the Task object.

Reply:
> You could comment out the sk.stop() above, and verify that the assertion fails.

It fails, as expected :)